test/extended/kubevirt/util.go

```diff
@@ -0,0 +1,317 @@
+	_, err := serviceClient.Create(context.Background(), service, metav1.CreateOptions{})
+	expectNoError(err)
+	expectNoError(exutil.WaitForEndpoint(client, namespace, serviceName))
+	service, err = serviceClient.Get(context.Background(), serviceName, metav1.GetOptions{})
+	expectNoError(err)
+
+	return int32(servicePort)
+
+}
+
+func checkConnectivityToHostWithCLI(f *e2e.Framework, oc *exutil.CLI, nodeName string, podName string, host string, timeout time.Duration, hostNetwork bool) error {
+	namespace := f.Namespace.Name
+
+	e2e.Logf("Creating an exec pod on node %v", nodeName)
+	execPod := pod.CreateExecPodOrFail(f.ClientSet, namespace, fmt.Sprintf("execpod-sourceip-%s", nodeName), func(pod *corev1.Pod) {
+		pod.Spec.NodeName = nodeName
+		pod.Spec.HostNetwork = hostNetwork
+	})
+	defer func() {
+		e2e.Logf("Cleaning up the exec pod")
+		err := f.ClientSet.CoreV1().Pods(namespace).Delete(context.Background(), execPod.Name, metav1.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	}()
+
+	var stdout string
+	e2e.Logf("Waiting up to %v to wget %s", timeout, host)
+	cmd := fmt.Sprintf("wget -T 30 -qO- %s", host)
+	var err error
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(2) {
+		stdout, err = oc.Run("exec").Args(execPod.Name, "-n", namespace, "--", "/bin/sh", "-x", "-c", cmd).Output()
+		if err != nil {
+			e2e.Logf("got err: %v, retry until timeout", err)
+			continue
+		}
+		// Need to check output because wget -q might omit the error.
+		if strings.TrimSpace(stdout) == "" {
+			e2e.Logf("got empty stdout, retry until timeout")
+			continue
+		}
+		break
+	}
+	return err
+}
+
+func getSchedulableNode(serverFramework *e2e.Framework) (string, error) {
+
+	// Any node is fine to schedule the server on.
+	nodes, err := e2enode.GetReadySchedulableNodes(serverFramework.ClientSet)
+	if err != nil {
+		e2e.Logf("Unable to get schedulable nodes due to %v", err)
+		return "", err
+	}
+	// GetReadySchedulableNodes guarantees us the list is not empty
+	// Any schedulable node will work for the server in this case.
+	return nodes.Items[0].Name, nil
+
+}
+
+func getKubeVirtPodFromGuestNode(framework *e2e.Framework, nodeName string) (*corev1.Pod, error) {
+
+	podList, err := framework.ClientSet.CoreV1().Pods(framework.Namespace.Name).List(context.Background(), metav1.ListOptions{})
+	Expect(err).To(Succeed())
+	for _, pod := range podList.Items {
+		if strings.Contains(pod.Name, nodeName) {
+			return &pod, nil
+		}
+	}
+
+	return nil, fmt.Errorf("kubevirt node's infra pod not found")
+}
+
+func getSchedulableInfraNode(framework *e2e.Framework, serverInfraNode string) (string, error) {
+
+	// find an infra node to schedule the client on
+	nodes, err := e2enode.GetReadySchedulableNodes(framework.ClientSet)
+	if err != nil {
+		e2e.Logf("Unable to get schedulable nodes due to %v", err)
+		return "", err
+	}
+
+	for _, node := range nodes.Items {
+		if node.Name != serverInfraNode {
+			return node.Name, nil
```

Comment:
This is returning the name of the first node that _isn't_ `serverInfraNode`, which... is not what the function name sounds like it is supposed to be doing?

Reply:
We want to get a schedulable infra node for the client pod, that is different than the node the server pod is running on.
note the function name implies we want to get a `SchedulableInfraNode`, and we're checking that node is not the infra node that running the **server** [pod]